ui/views.py

```diff
@@ -56,6 +58,26 @@ def default_js_settings(request):
     }
 
 
+def conditional_response(view, *args, **kwargs):
```

Comment:
I'm not sure that this function should know about `TemplateView.get`. Can we have the calling function pass in a `view_func` and execute and return that value instead?

Reply:
How about replacing it with this, tried it and it works:
```
    context = view.get_context_data(video, **kwargs)
    return view.render_to_response(context)
```